### ISSUE TITLE
Systematic crash upon updated vendor list

### ISSUE THREAD

**Issue by PasqualePuzio** [Fidzup/fidzup-gdpr-cmp-android#2]
Hi,

we've identified a major issue occurring to the vast majority of our users (about 50%).

This issue is causing the app to crash in the attempt of updating the consent string after the vendor list has been updated (the issue surfaced today, I suppose the IAB has just published a new version of the vendor list).

Please find attached the stacktrace and the information about the crash.

Thanks

[com.puzio.fantamaster_issue_crash_7F6CD67403D100015FCB29D6D84EABC3_DNE_0_v2.txt](https://github.com/Fidzup/fidzup-gdpr-cmp-android/files/3472296/com.puzio.fantamaster_issue_crash_7F6CD67403D100015FCB29D6D84EABC3_DNE_0_v2.txt)

**Comment by PasqualePuzio** [comment]
Any news on this issue? Thanks

**Comment by sinfomicien** [comment]
Yep, i found the bug... It happened in some situations when the VendorList is updated, and conflict with the one in cache locally.
I already wrote a patch, and actually i'm tested it on some production application. If it's correct, i hope to publish a new version end of the week, or at least next week.

**Comment by PasqualePuzio** [comment]
Great news! Looking forward to integrating your CMP again. Thanks

**Comment by PasqualePuzio** [comment]
Hi @sinfomicien sorry to bother you again on the same issue: is there any news? When can e expect a new release of the CMP? Thanks

**Comment by PasqualePuzio** [comment]
Is this library ever going to be updated? Should we expect a fix soon or we should consider moving to an alternative? Thanks
